velox/common/memory/HashStringAllocator.cpp

```diff
@@ -147,14 +147,24 @@ HashStringAllocator::Position HashStringAllocator::finishWrite(
 }
 
 void HashStringAllocator::newSlab(int32_t size) {
+  char* run = nullptr;
+  uint64_t available = 0;
   int32_t needed = std::max<int32_t>(
       bits::roundUp(
           size + 2 * sizeof(Header), memory::AllocationTraits::kPageSize),
       kUnitSize);
-  pool_.newRun(needed);
-  auto run = pool_.firstFreeInRun();
-  auto available = pool_.availableInRun() - sizeof(Header);
-
+  auto pagesNeeded = memory::AllocationTraits::numPages(needed);
+  if (pagesNeeded > pool()->largestSizeClass()) {
```

Comment:
We probably need some metrics or WARNING log here as `HashStringAllocator` usually is not meant for large contiguous allocations